test/interface.spec.js

```diff
@@ -0,0 +1,179 @@
+      // files.catPullStream
+      // TODO not implemented in go-ipfs yet
+      'should export a chunk of a file',
+      'should export a chunk of a file in a Pull Stream',
+      'should export a chunk of a file in a Readable Stream',
+      // files.get
+      // FIXME https://github.com/ipfs/js-ipfs-api/issues/339
+      isNode ? null : 'should get a directory'
+    ]
+  })
+
+  tests.key(defaultCommonFactory, {
+    skip: [
+      // key.export
+      // TODO not implemented in go-ipfs yet
+      'export',
+      // key.import
+      // TODO not implemented in go-ipfs yet
+      'import'
+    ]
+  })
+
+  tests.ls(defaultCommonFactory, {
+    skip: [
+      // lsPullStream
+      // FIXME https://github.com/ipfs/js-ipfs-api/issues/339
+      isNode ? null : 'should pull stream ls with a base58 encoded CID',
+      // lsReadableStream
+      // FIXME https://github.com/ipfs/js-ipfs-api/issues/339
+      isNode ? null : 'should readable stream ls with a base58 encoded CID',
+      // ls
+      // FIXME https://github.com/ipfs/js-ipfs-api/issues/339
+      isNode ? null : 'should ls with a base58 encoded CID'
+    ]
+  })
+
+  tests.miscellaneous(defaultCommonFactory, {
+    skip: [
+      // stop
+      // FIXME go-ipfs returns an error https://github.com/ipfs/go-ipfs/issues/4078
+      'should stop the node'
+    ]
+  })
+
+  tests.object(defaultCommonFactory)
+
+  tests.pin(defaultCommonFactory)
+
+  tests.ping(defaultCommonFactory)
+
+  tests.pubsub(CommonFactory.create({
+    spawnOptions: {
+      args: ['--enable-pubsub-experiment'],
+      initOptions: { bits: 1024 }
+    }
+  }), {
+    skip: isNode ? [
+      // pubsub.subscribe
+      // FIXME https://github.com/ipfs/interface-ipfs-core/pull/188#issuecomment-354673246
+      // and https://github.com/ipfs/go-ipfs/issues/4778
+      isWindows ? 'should send/receive 100 messages' : null,
+      isWindows ? 'should receive multiple messages' : null
+    ] : true
+  })
+
+  tests.repo(defaultCommonFactory)
+
+  tests.stats(defaultCommonFactory)
+
+  tests.swarm(CommonFactory.create({
+    createSetup ({ ipfsFactory, nodes }) {
+      return callback => {
+        callback(null, {
+          spawnNode (repoPath, config, cb) {
+            if (typeof repoPath === 'function') {
+              cb = repoPath
+              repoPath = undefined
+            }
+
+            if (typeof config === 'function') {
+              cb = config
+              config = undefined
+            }
+
+            const spawnOptions = { repoPath, config, initOptions: { bits: 1024 } }
+
+            ipfsFactory.spawn(spawnOptions, (err, _ipfsd) => {
+              if (err) {
+                return cb(err)
+              }
+
+              nodes.push(_ipfsd)
+              cb(null, IPFSApi(_ipfsd.apiAddr))
+            })
+          }
+        })
+      }
+    }
+  }))
+
+  tests.types(defaultCommonFactory, { skip: true })
+
+  tests.util(defaultCommonFactory, { skip: true })
```

Comment:
Why skip?

Reply:
These were not being run before and they fail right now - I'll add a comment